pkg/sdk/external_volumes_def.go

```diff
@@ -0,0 +1,161 @@
+	TextAssignment("AZURE_TENANT_ID", g.ParameterOptions().SingleQuotes().Required()).
+	TextAssignment("STORAGE_BASE_URL", g.ParameterOptions().SingleQuotes().Required())
+
+var storageLocationDef = g.NewQueryStruct("ExternalVolumeStorageLocation"). // Can't name StorageLocation due to naming clash with type in storage integration
+										OptionalQueryStructField(
+		"S3StorageLocationParams",
+		externalS3StorageLocationDef,
+		g.ListOptions().Parentheses().NoComma(),
+	).
+	OptionalQueryStructField(
+		"GCSStorageLocationParams",
+		externalGCSStorageLocationDef,
+		g.ListOptions().Parentheses().NoComma(),
+	).
+	OptionalQueryStructField(
+		"AzureStorageLocationParams",
+		externalAzureStorageLocationDef,
+		g.ListOptions().Parentheses().NoComma(),
+	).
+	WithValidation(g.ExactlyOneValueSet, "S3StorageLocationParams", "GCSStorageLocationParams", "AzureStorageLocationParams")
+
+var ExternalVolumesDef = g.NewInterface(
+	"ExternalVolumes",
+	"ExternalVolume",
+	g.KindOfT[AccountObjectIdentifier](),
+).
+	CreateOperation(
+		"https://docs.snowflake.com/en/sql-reference/sql/create-external-volume",
+		g.NewQueryStruct("CreateExternalVolume").
+			Create().
+			OrReplace().
+			SQL("EXTERNAL VOLUME").
+			IfNotExists().
+			Name().
+			ListAssignment("STORAGE_LOCATIONS", "ExternalVolumeStorageLocation", g.ParameterOptions().Parentheses().Required()).
+			OptionalBooleanAssignment("ALLOW_WRITES", nil).
+			OptionalComment().
+			WithValidation(g.ConflictingFields, "OrReplace", "IfNotExists").
+			WithValidation(g.ValidIdentifier, "name"),
+		storageLocationDef,
+	).
+	AlterOperation(
+		"https://docs.snowflake.com/en/sql-reference/sql/alter-external-volume",
+		g.NewQueryStruct("AlterExternalVolume").
+			Alter().
+			SQL("EXTERNAL VOLUME").
+			IfExists().
+			Name().
+			OptionalTextAssignment("REMOVE STORAGE_LOCATION", g.ParameterOptions().SingleQuotes().NoEquals()).
+			OptionalQueryStructField(
+				"Set",
+				g.NewQueryStruct("AlterExternalVolumeSet").
+					OptionalBooleanAssignment("ALLOW_WRITES", g.ParameterOptions()).
+					OptionalComment(),
+				g.KeywordOptions().SQL("SET"),
+			).
+			OptionalQueryStructField(
+				"AddStorageLocation",
+				storageLocationDef,
+				g.ParameterOptions().SQL("ADD STORAGE_LOCATION"),
+			).
+			WithValidation(g.ExactlyOneValueSet, "RemoveStorageLocation", "Set", "AddStorageLocation").
+			WithValidation(g.ValidIdentifier, "name"),
+	).
+	DropOperation(
+		"https://docs.snowflake.com/en/sql-reference/sql/drop-external-volume",
+		g.NewQueryStruct("DropExternalVolume").
+			Drop().
+			SQL("EXTERNAL VOLUME").
+			IfExists().
+			Name().
+			WithValidation(g.ValidIdentifier, "name"),
+	).
+	DescribeOperation(
+		g.DescriptionMappingKindSlice,
+		"https://docs.snowflake.com/en/sql-reference/sql/desc-external-volume",
+		g.DbStruct("externalVolumeDescRow").
+			Field("parent_property", "string").
```

Comment:
nit-pick: we have helper methods for types, check e.g. https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/pkg/sdk/cortex_search_services_def.go#L62